actionpack/lib/action_controller/metal/renderers.rb

```diff
@@ -103,17 +123,88 @@ def self.remove(key)
       remove_method(method_name) if method_defined?(method_name)
     end
 
+    # Serializers define a method called within a renderer specific to
+    # transforming the object into a mime-compatible type.
+    #
+    # The separation of serialization from rendering allows
+    # composing the Renderer behavior of two methods, e.g.
+    # +_render_with_renderer_json+ and +_serialize_with_serializer_json+,
+    # rather than requiring one to define a method +_render_with_renderer_json+
+    # in a subclass and optionally call super on it.
+    #
+    # A principal benefit of this approach is that it promotes serialization on an object
+    # to a clearly-defined public interface, rather than requiring one to understand that
+    # calling, e.g. +render json: object+ calls +_render_to_body_with_renderer(options)+
+    # which calls +_render_with_renderer_#{key}+ where key is +json+, which is the method
+    # defined by calling +ActionController::Renderers.add :json+.
+    #
+    # ActiveModel::Serializers, for example, which has relied upon defining
+    # +_render_option_json+ and +_render_with_renderer_json+ in the controller, and calling +super+
+    # on the serialized object, now only needs to call
+    # +ActionController::Serializers.remove_serializer :json+ and define a new serializer with
+    # +ActionController::Serializers.add_serializer json do |json, options|+.
+    #
+    # See https://groups.google.com/forum/#!topic/rubyonrails-core/K8t4-DZ_DkQ/discussion for
+    # more background information.
```

Comment:
What do you think of this documentation?

```ruby
    # Serializers define a method called within a renderer specific to
    # transforming the object into a mime-compatible type.
    #
    # The separation of serialization from rendering allows
    # composing the Renderer behavior of two methods, e.g.
    # +_render_with_renderer_json+ and +_serialize_with_serializer_json+,
    # rather than requiring one to define a method +_render_with_renderer_json+
    # in a subclass and optionally call super on it.
    #
    # A principal benefit of this approach is that it promotes serialization on an object
    # to a clearly-defined public interface, rather than requiring one to understand that
    # calling, e.g. +render json: object+ calls +_render_to_body_with_renderer(options)+
    # which calls +_render_with_renderer_#{key}+ where key is +json+, which is the method
    # defined by calling +ActionController::Renderers.add :json+.
    #
    # ActiveModel::Serializers, for example, which has relied upon defining
    # +_render_option_json+ and +_render_with_renderer_json+ in the controller, and calling +super+
    # on the serialized object, now only needs to call
    # +ActionController::Serializers.remove_serializer :json+ and define a new serializer with
    # +ActionController::Serializers.add_serializer json do |json, options|+.
    #
    # See https://groups.google.com/forum/#!topic/rubyonrails-core/K8t4-DZ_DkQ/discussion for
    # more background information.
```